src/python_testing/TestSpecParsingSupport.py

```diff
@@ -106,6 +221,80 @@ def test_write_optional(self):
+        asserts.assert_equal(len(ids_by_name), 1, "Unexpected number of IDs per name")
+        asserts.assert_equal(len(problems), 0, "Unexpected number of problems")
+        asserts.assert_equal(ids_by_name["Test Derived"], 0xFFFF, "Test derived name not added to IDs")
+
+        asserts.assert_true(0xFFFF in clusters, "Derived ID not found in clusters")
+        asserts.assert_equal(set(clusters[0xFFFF].attributes.keys()), set(
+            [0, 2, 3] + expected_global_attrs), "Unexpected attribute list")
+        asserts.assert_equal(set(clusters[0xFFFF].accepted_commands.keys()), set([]), "Unexpected accepted commands")
+        asserts.assert_equal(set(clusters[0xFFFF].generated_commands.keys()), set([]), "Unexpected generated commands")
+
+        asserts.assert_true("Test Base" in derived_clusters, "Base ID not found in derived clusters")
+        asserts.assert_equal(set(derived_clusters["Test Base"].attributes.keys()), set(
+            [0, 1, 2, 3] + expected_global_attrs), "Unexpected attribute list")
+        asserts.assert_equal(set(derived_clusters["Test Base"].accepted_commands.keys()), set([0]), "Unexpected accepted commands")
+        asserts.assert_equal(set(derived_clusters["Test Base"].generated_commands.keys()),
+                             set([1]), "Unexpected generated commands")
+        asserts.assert_equal(str(derived_clusters["Test Base"].accepted_commands[0].conformance),
+                             "M", "Unexpected conformance on base accepted command")
+        asserts.assert_equal(str(derived_clusters["Test Base"].generated_commands[1].conformance),
+                             "M", "Unexpected conformance on base generated command")
+
+        asserts.assert_equal(len(derived_clusters["Test Base"].unknown_commands),
+                             0, "Unexpected number of unknown commands in base")
+        asserts.assert_equal(len(clusters[0xFFFF].unknown_commands), 2, "Unexpected number of unknown commands in derived cluster")
+
+        combine_derived_clusters_with_base(clusters, derived_clusters, ids_by_name)
+        # Ensure the base-only attribute (1) was added to the derived cluster
+        asserts.assert_equal(set(clusters[0xFFFF].attributes.keys()), set(
+            [0, 1, 2, 3] + expected_global_attrs), "Unexpected attribute list")
+        # Ensure the conformance overrides from the derived cluster are on the attributes
+        asserts.assert_equal(str(clusters[0xFFFF].attributes[0].conformance), "M", "Unexpected conformance on attribute 0")
+        asserts.assert_equal(str(clusters[0xFFFF].attributes[1].conformance), "M", "Unexpected conformance on attribute 0")
+        asserts.assert_equal(str(clusters[0xFFFF].attributes[2].conformance), "X", "Unexpected conformance on attribute 1")
+        asserts.assert_equal(str(clusters[0xFFFF].attributes[3].conformance), "X", "Unexpected conformance on attribute 1")
```

Comment:
should this possible be?

```suggestion
        asserts.assert_equal(str(clusters[0xFFFF].attributes[1].conformance), "M", "Unexpected conformance on attribute 1")
        asserts.assert_equal(str(clusters[0xFFFF].attributes[2].conformance), "X", "Unexpected conformance on attribute 2")
        asserts.assert_equal(str(clusters[0xFFFF].attributes[3].conformance), "X", "Unexpected conformance on attribute 3")
```